### ISSUE TITLE
Add ability to query for ingress rules to security group resources

### ISSUE THREAD

**Issue by clintoncwolfe** [chef/inspec-aws#137]
I'd like to be able to do something like:
```
describe aws_ec2_security_group('sg-12345678') do
  its('ingress_rule.count') { should be 2 }
  its('ingress_rules.last') { should include(:action => 'REJECT', :source =>'0.0.0.0/0') }
  it { should_not be_open_to_the_world_on_port(22) }
  its('ingress_rules.first') { should include(source: 'sg-87654321', port_from: 22, port_to: 22) }
end
```

So, for the singular resource, I was thinking of having the `ingress_rules` property be an array of hashes, sorted by rule order.  Based on that property, I'd also add a high-level matcher, `open_to_the_world_on_port?(N)`.

And for the plural case:
```
describe aws_ec2_security_groups.where { open_to_the_world_on_port(22) }  do
  it { should_not exist }
end
```

And perhaps:
```
describe aws_ec2_security_groups.where { blocks_all_ingress }  do
  it { should exist }
end
```

Refs #129

**Comment by aaronlippold** [comment]
@dromazmj and @hackershark are working. Look for PRs soon.

As a side note, from our research it doesn't seem like the API has the concept of a REJECT rule in a security_group. At least that is what they tell me. Do we need to rework what we want this to look like? 

`its('ingress_rules.last') { should include(:action => 'REJECT', :source =>'0.0.0.0/0') }`

**Comment by burtlo** [comment]
Does the order of the rule matter? The use of `first` and `last` seemed a little strange to me in this situation.

I started to think about how I would express the expectations for a simple inbound SSH and outbound anything and I thought this was working pretty well.

```ruby
describe aws_ec2_security_group(id: 'sg-b88c6dcf') do
  it { should exist }
  its('inbound.rules') { should permit({ protocol: :tcp, port: 22, source: '0.0.0.0/0' })
  }
  its('outbound.rules') { should permit_all_traffic }
  its('outbound.rules') { should permit(:all_traffic) } # This isn't implemented but would save me a matcher.
end
```

Here is the example of controls with the initial implementation of the matchers I wrote: https://github.com/burtlo/training-aws-security/blob/master/controls/aws_ec2_security_group.rb

Here is a rudimentary implementation of supporting those expectations: https://github.com/burtlo/inspec-aws/commit/81f4ce64b337468da0af9a4168745b67520d1045

I initially wanted to use `allow` but that is, of course, used by RSpec mock syntax. So I went with a `permit` matcher. But now that I think about it more, a matcher named `permit` is really generic so perhaps it should be called `permit_traffic`
